hedera-node/src/main/java/com/hedera/services/state/logic/StandardProcessLogic.java

```diff
@@ -86,42 +86,64 @@ public StandardProcessLogic(
 	public void incorporateConsensusTxn(SwirldTransaction platformTxn, Instant consensusTime, long submittingMember) {
 		try {
 			final var accessor = expandHandleSpan.accessorFor(platformTxn);
-			Instant effectiveConsensusTime = consensusTime;
-			if (accessor.canTriggerTxn()) {
-				final var offset = dynamicProperties.triggerTxnWindBackNanos();
-				effectiveConsensusTime = consensusTime.minusNanos(offset);
-			}
 
-			if (!invariantChecks.holdFor(accessor, effectiveConsensusTime, submittingMember)) {
+			if (!invariantChecks.holdFor(accessor, consensusTime, submittingMember)) {
 				return;
 			}
 
-			sigImpactHistorian.setChangeTime(effectiveConsensusTime);
-			expiries.purge(effectiveConsensusTime.getEpochSecond());
+			consensusTimeTracker.reset(consensusTime);
+
+			sigImpactHistorian.setChangeTime(consensusTime);
+			expiries.purge(consensusTime.getEpochSecond());
 			sigImpactHistorian.purge();
 			recordStreaming.resetBlockNo();
 
-			doProcess(submittingMember, consensusTime, effectiveConsensusTime, accessor);
+			doProcess(submittingMember, consensusTimeTracker.isFirstUsed()
```

Comment:
Isn't `consensusTimeTracker.isFirstUsed()` guaranteed to be `false` here?

(Or is this just defensive programming, in case someday that isn't guaranteed?)